### PR TITLE
Adding dash to allowed characters for the cluster to allow SDK being …

### PR DESCRIPTION
**Pull request by senickel**
…used with power-no cluster

### CODE REVIEW

File: cognite/client/_api_client.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def _get_base_url_with_base_path(self) -> str:
 
     def _is_retryable(self, method: str, path: str) -> bool:
         valid_methods = ["GET", "POST", "PUT", "DELETE", "PATCH"]
-        match = re.match("(?:http|https)://[a-z\d.:]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
+        match = re.match("(?:http|https)://[a-z\d.:-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
```
Comment (sanderland) [inline cognite/client/_api_client.py:193]:
```suggestion
         match = re.match("(?:http|https)://[a-z\d.:-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
```

**Comment by me-ydv-5** [comment]
Can you please add a corresponding test and changelog and version bump? :)

**Review by sanderland — CHANGES_REQUESTED** [review]
also, a test would be nice

File: cognite/client/_api_client.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ def _get_base_url_with_base_path(self) -> str:
 
     def _is_retryable(self, method: str, path: str) -> bool:
         valid_methods = ["GET", "POST", "PUT", "DELETE", "PATCH"]
-        match = re.match("(?:http|https)://[a-z\d.:]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
+         match = re.match("(?:http|https)://[a-z\d.:-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
```
Comment (sanderland) [inline cognite/client/_api_client.py:193]:
```suggestion
        match = re.match("(?:http|https)://[a-z\d.:\-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
``

**Review by sanderland — COMMENTED** [review]
to avoid - from turning into a character range, and indent seems off

File: cognite/client/_api_client.py [hunk 3]
```diff
@@ -190,7 +190,8 @@ def _get_base_url_with_base_path(self) -> str:
 
     def _is_retryable(self, method: str, path: str) -> bool:
         valid_methods = ["GET", "POST", "PUT", "DELETE", "PATCH"]
-         match = re.match("(?:http|https)://[a-z\d.:-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
+        match = re.match("(?:http|https)://[a-z\d.:\-]+(?:/api/(?:v1|playground)/projects/[^/]+)?(/[^\?]+)?(\?.+)?", path)
+``
```
Comment (sanderland) [inline cognite/client/_api_client.py:194]: ```suggestion```

Comment (sanderland) [inline cognite/client/_api_client.py:194]: err but not these, oops

**Review by sanderland — APPROVED** [review]

**Pull request merged**